src/Backends/Helmholtz/VLERoutines.cpp

```diff
@@ -554,6 +554,52 @@ void SaturationSolvers::saturation_PHSU_pure(HelmholtzEOSMixtureBackend& HEOS, C
                                        info.c_str(), specified_value, error));
         }
     } while (error > 1e-9);
+    // Recalculate error
+    // The result has changed since the last error calculation.
+    // In rare scenarios, the final step can become unstable due to solving a singular
+    // J matrix. This final error check verifies that the solution is still good.
+    // Furthermore, the forced phase of SatL and SatV may have caused errors. We will recalculate them without this assumption.
```

Comment:
Does the singular J matrix still crop up with the fixed ancillary? I'm wondering if this fix is still needed?

Reply:
I never actually checked whether it is still needed. I figure that it is a cheap enough check that it is probably worth leaving in.

Reply:
Okay, great. I'm satisfied with that.

Reply:
@msaitta-mpr @ibell, so what are the next steps to get this merged?

Reply:
If everyone agrees we are ready I can merge it